app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,29 @@
+class VotesController < ApplicationController
+  before_action :ensure_current_user
+
+  def create
+    if params[:voteable_type] == 'Question'
```

Comment:
This is not ideal because you will need to keep changing this method every time you add a new type of vote.  You have two choices:  handle all votes in the votes controller or handle votes in each controller that can accept votes.  If you are going to handle it in the votes controller, you may want to check out this Railscast:

http://railscasts.com/episodes/154-polymorphic-association

This would mean that after this controller is refactored, you would have something like this:
```
    @voteable = find_voteable
    @voteable.votes.create(params[:vote])
    redirect_to :back
```